framework/src/kernels/MaterialDerivativeTestKernel.C

```diff
@@ -0,0 +1,66 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "MaterialDerivativeTestKernel.h"
+
+
+template<>
+InputParameters validParams<MaterialDerivativeTestKernel>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addClassDescription("Class used for testing derivatives of a material property.");
+  params.addRequiredParam<MaterialPropertyName>("material_property", "Name of material property for which derivatives are to be tested.");
+  params.addRequiredCoupledVar("vars", "List of nonlinear variables.");
+
+  return params;
+}
+
+MaterialDerivativeTestKernel::MaterialDerivativeTestKernel(const InputParameters & parameters) :
+    DerivativeMaterialInterface<Kernel>(parameters),
+    _p_name(getParam<MaterialPropertyName>("material_property")),
+    _p(getMaterialPropertyByName<Real>(_p_name))
```

Comment:
remove `_p_name` and use _p(getMaterialProperty<Real>("material_property"))` here (and similarly for the derivatives)